src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -610,6 +612,9 @@ private static bool StringRequiresHtmlDecoding(string s)
         // Internal struct to facilitate URL decoding -- keeps char buffer and byte buffer, allows appending of either chars or bytes
         private struct UrlDecoder
         {
+            public bool _containsUnsafe;
+            public bool _containsSpaces;
```

Comment:
I agree with moving them out of UrlDecoder.
I would like to keep the fields `containsUnsafe` and `containsSpaces` as separate variables, as I'm working on a PR that's gonna heavily optimize `UrlEncode` and `UrlDecode` for strings that only need space encoding/decoding. Is that OK?